static/js/views/settings.js

```diff
@@ -865,17 +867,17 @@ const SettingsScreen = {
     this.hideNetworkElements();
     this.elements.network.main.classList.remove('hidden');
 
-    // set default values in case things go bad
-    this.elements.network.client.ethernet.netmask.value = '255.255.255.0';
-
     this.elements.network.client.ethernet.main.classList.remove('hidden');
 
     API.getLanSettings()
       .then((body) => {
         this.elements.network.client.ethernet.mode.value = body.mode || 'dhcp';
-        this.elements.network.client.ethernet.ip.value = body.ipdaddr || '';
-        this.elements.network.client.ethernet.netmask.value = body.netmask || '255.255.255.0';
-        this.elements.network.client.ethernet.gateway.value = body.gateway || '';
+        if (body.mode === 'static') {
+          this.ethernetEls.forEach((el) => el.classList.remove('hidden'));
```

Comment:
I'm sure there's a more efficient way of doing this, like grouping the inputs into a fieldset and showing/hiding the fieldset instead, but I think this is OK if you'd rather not get into modifying the HTML structure.